myoFlex/myoFlex.ino

```diff
@@ -1,69 +1,131 @@
+  pointer.attach(pointerPin, min, max);
+  middle.attach(middlePin, min, max);
+  ring.attach(ringPin, min, max);
+  pinky.attach(pinkyPin, min, max);
+  Serial.println("Starting");       //makes the hand relax all of the way
+  handPosition(max,max,max,max,max);
+  trigger = setTrigger();
+}
+
+//setup starts by linking the Servo objects to each pin and detailing the limits per each argument. Then it opens the hand for a default position and will call the setTrigger function. Once it's collected, the setup is complete
+
+void loop() {
+  Serial.println(analogRead(myoIn));
+  if(check(currentVoltage = analogRead(myoIn))){//collects voltage, then assigns to currentVoltage, then passes currentVoltage to check function, then checks the check return value to see if it's higher than trigger value
+    state++; //increases our state
+    state = state%4; //ensoures our only values are between 0-3 and loops back if otherwise
+    Serial.print("state is ");
+    Serial.println(state);
+    //Serial.println(currentVoltage);        //prints in the serial moniter
+    if(state == 0){
+      Serial.println("Open");
+      handPosition(max,max,max,max,max);
+    }
+    else if(state == 1){
+      Serial.println("Point");
+      handPosition(max,min,max,max,max);
+    }
+    else if(state == 2){
+      Serial.println("Pinch");
+      handPosition(min,min,min,max,max);
+    }
+    else if(state == 3){
+      Serial.println("Fist");
+      handPosition(min,min,min,min,min);
+    }
+    else {
+      state = 1; //edgecase if logic fail
+      handPosition(max,max,max,max,max);
+    }
+  }
+}
+
+//the loop function will continously check the users input and assign it to currentVoltage, then it's verified through the check function to see if its higher than the set trigger value from setup, if so then it will switch between the four hand positions in a looping structure
+
+int setTrigger(){
+  float average = 0;
+  int i, j;
+  do{
+    average = 0;//resets collection value if it failed prior
+    for(j = 1; j <= 3; j++){//loops to collect the three impulses
+      Serial.print("Reading impulse ");
+      Serial.print(j);
+      Serial.println(" in the next 3 seconds");
+      for(i = 3; i != 0; i--){//counts down 3 seconds
+        Serial.println(i);
+        delay(1000);
+      }
+      Serial.println("Reading");
+      average = average + analogRead(myoIn);//adds up collected values with previous values
+      Serial.println(average/3);
+    }
+  } while((average/3) < 200);//if it's less than 200 than we reattempt collection
+  return (int)(average/3);//returns average upon success
+}
+
+//setTrigger collects 3 inputs by the user with a 3 second delay between each one. It calculates the average and if it's lower than 200, it's an assumed error and will reset the collection. Once completed it returns the average
+
+int check(int voltage){
+  if(voltage > trigger){
+    return 1;
+  }
+  return 0;
+}
```

Comment:
Also, I think we should have a decimal multiplier in front of the trigger. The person shouldn't have to hit the peak signal every time, maybe just 60% or 80% of it.

Thanks. This looks great.